tests/unit/test_table.py

```diff
@@ -1050,6 +1050,12 @@ def test_mview_refresh_interval(self):
         table.mview_refresh_interval = None
         self.assertIsNone(table.mview_refresh_interval)
 
+    def test_mview_allow_non_incremental_definition(self):
+        table = self._make_one()
+        self.assertIsNone(table.mview_allow_non_incremental_definition)
+        table.mview_allow_non_incremental_definition = True
+        self.assertTrue(table.mview_allow_non_incremental_definition)
```

Comment:
Gonna mirror the test above (`test_mview_enable_refresh`) and cycle through all the options:

```suggestion
        self.assertTrue(table.mview_allow_non_incremental_definition)
        table.mview_allow_non_incremental_definition = False
        self.assertFalse(table.mview_allow_non_incremental_definition)
        table.mview_allow_non_incremental_definition = None
        self.assertIsNone(table.mview_allow_non_incremental_definition)
```